tools/hangar/__snapshots__/invalid.ts.snap

```diff
@@ -365,11 +365,11 @@ Duration <DURATION>"
 `;
 
 exports[`bring_non_std_construct.test.w 1`] = `
-"error: Expected 1 positional argument(s) but got 0
+"error: Expected between 1 and 2 arguments but got 0
   --> ../../../examples/tests/invalid/bring_non_std_construct.test.w:8:1
   |
 8 | new cdktf.S3Backend();
-  | ^^^^^^^^^^^^^^^^^^^^^ Expected 1 positional argument(s) but got 0
+  | ^^^^^^^^^^^^^^^^^^^^^ Expected between 1 and 2 arguments but got 0
```

Comment:
The new error message here seems a bit misleading to me. We actually expect either 2 pos argument or 1 pos argument N named argument fitting `S3BackendConfig`'s fields.
Granted, the old message was misleading too.

Reply:
What would be a better error message in such a case? Maybe we can use a different message depending on whether the last param is a struct?

Reply:
How does this work in more complex cases? Suppose there are multiple named arguments:

```js
struct ComputeOptions {
  opt1: str;
  opt2: str;
  opt3: str?;
  opt4: str?;
}

let compute = (arg1: num, arg2: num, opts: ComputeOptions) => {
  // implementation
};
compute(123); // error (a)
compute(123, 456, opt2: "yes", opt1: "purple"); // ok!
```

What should the error be at (a)? The message we have today feels sort of incomplete:
`Expected 2 positional argument(s) but got 1`

One option we could consider is to provide the user a full list of the names of all missing arguments:
`Missing 1 positional argument ("arg2") and 2 named arguments ("opt1", "opt2")`
or
`Expected 2 positional arguments and 2 named arguments, but only received 1 positional argument`

Just brainstorming some ideas 😄 

----

BTW - in the future it should be possible to add "hints" to errors, so e.g. we could suggest what the correct syntax look like:

```
hint: I expected the function call to look like
    compute(arg1, arg2, opt1: exp1, opt2: exp2)
```

Reply:
Updated the error to give more details about variadic and named arguments. Also checking that not both a positional arg and named args are given for the last parameter. I like the idea of the hint but not sure if it belongs in this pr as the original goal was to fix variadic type checks and arity check and thats more related to named args and more of a nice addition than a fix.